### PR TITLE
reconcile chart status

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: 
- https://github.com/giantswarm/giantswarm/issues/4077
- https://github.com/giantswarm/giantswarm/issues/4602

Reconcile charts status, ~and recover from failed charts deployment~.

### CODE REVIEW

File: service/controller/chartconfig/v5/resource/chart/update.go [hunk 1]
```diff
@@ -93,7 +93,9 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if the chart has to be updated")
 
-	if !currentChartState.IsEmpty() && isChartModified(currentChartState, desiredChartState) {
+	isModified := !currentChartState.IsEmpty() && isChartModified(currentChartState, desiredChartState)
+	isWrongStatus := currentChartState.ReleaseStatus != desiredChartState.ReleaseStatus && !isChartInTransitionState(currentChartState)
+	if isModified || isWrongStatus {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "the chart has to be updated")
```
Comment (kopiczko) [inline service/controller/chartconfig/v5/resource/chart/update.go]: Does that mean that release in state `PENDING_UPGRADE` or `DELETING` should be updated?

Comment (kopiczko) [reply service/controller/chartconfig/v5/resource/chart/update.go]: Sorry it's other way around `!`.

**Comment by kopiczko** [comment]
FMI: How does it "recover from failed charts deployment"?

**Comment by TheoBrigitte** [comment]
> FMI: How does it "recover from failed charts deployment"?

It actually doesn't. I did a lame copy of the previous PR message.

**Pull request merged**